common/Resources/ti.internal/kernel/android/nativemodule.js

```diff
@@ -70,14 +70,29 @@ export default function NativeModuleBootstrap(global, kroll) {
 
 	NativeModule.prototype.compile = function () {
 
-		let source = NativeModule.getSource(this.id);
-		source = NativeModule.wrap(source);
-
 		// All native modules have their filename prefixed with ti:/
 		const filename = `ti:/${this.filename}`;
 
-		const fn = runInThisContext(source, filename, true);
-		fn(this.exports, NativeModule.require, this, this.filename, null, global.Ti, global.Ti, global, kroll);
+		let source = NativeModule.getSource(this.id);
+
+		if (OS_IOS) {
+			source = NativeModule.wrap(source);
+
+			const f = evaluate.runInThisContext(source, filename, true);
+			f(this.exports, NativeModule.require, this, this.filename, null, global.Ti, global.Ti, global, kroll);
+
+		} else if (OS_ANDROID) {
+			const result = evaluate.runAsModule(source, filename, {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/kernel/android/nativemodule.js;
  Line: 85;
-->

- :warning: common/Resources/ti.internal/kernel/android/nativemodule.js line 85 – 'result' is assigned a value but never used. (no-unused-vars)